deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LayerNormalizationWholeMinibatch.java

```diff
@@ -0,0 +1,21 @@
+package org.deeplearning4j.nn.layers.recurrent;
+
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+/**
+ * Created by fac2003 on 4/6/17.
+ */
+public class LayerNormalizationWholeMinibatch implements LayerNormalization {
+    @Override
+    public void normalize(INDArray activations) {
+        // normalize across entires mini-batch:
+        Number stdev = activations.stdNumber();
+        activations.subi(Nd4j.mean(activations));
+        final double stdNumber = stdev.doubleValue();
+        if (!Double.isNaN(stdNumber)) {
```

Comment:
So this scares me a bit. If we add an epsilon instead we can do some optimizations later to make it inline.Reduction algorithms (sums,..) are expensive.